lib/v2/components/dashboard/components/currency_info_card.dart

```diff
@@ -0,0 +1,87 @@
+import 'package:seeds/v2/design/app_theme.dart';
+import 'package:seeds/v2/models/token_model.dart';
+
+class CurrencyInfoCard extends StatelessWidget {
+  final TokenModel token;
+  final String backgroundImage;
+  final String logo;
+  final String balanceSubTitle;
+  final String fiatBalance;
+  final double? cardWidth;
+  final double? cardHeight;
+  final Color? textColor;
+
+  const CurrencyInfoCard({
+    Key? key,
+    required this.token,
+    required this.backgroundImage,
+    required this.logo,
+    required this.balanceSubTitle,
+    required this.fiatBalance,
+    this.cardWidth,
+    this.cardHeight,
+    this.textColor,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocProvider<BalanceBloc>(
+      create: (context) => BalanceBloc(token: token)..add(OnBalanceUpdate()),
+      child: BlocBuilder<BalanceBloc, BalanceState>(
+        builder: (context, state) {
+          return Container(
+              width: cardWidth,
+              height: cardHeight,
+              decoration: BoxDecoration(
+                  borderRadius: BorderRadius.circular(24),
+                  image: DecorationImage(image: AssetImage(backgroundImage), fit: BoxFit.fill)),
+              child: Stack(children: <Widget>[
+                Padding(
+                  padding: const EdgeInsets.only(top: 20, right: 20, left: 20),
+                  child: Column(
+                    crossAxisAlignment: CrossAxisAlignment.start,
+                    children: <Widget>[
+                      Row(
+                        children: <Widget>[
+                          Expanded(
+                              child: Text(
+                            state.token.name,
+                            style: Theme.of(context).textTheme.headline7.copyWith(color: textColor),
+                          )),
+                          Container(
+                              width: 42,
+                              height: 42,
+                              decoration: BoxDecoration(
+                                  borderRadius: BorderRadius.circular(8),
+                                  image: DecorationImage(image: AssetImage(logo), fit: BoxFit.fill))),
+                        ],
+                      ),
+                      const SizedBox(
+                        height: 50,
+                      ),
+                      Text("Balance", style: Theme.of(context).textTheme.subtitle2!.copyWith(color: textColor)),
+                      const SizedBox(
+                        height: 6,
+                      ),
+                      Text(state.displayString(), style: Theme.of(context).textTheme.headline5!.copyWith(color: textColor)),
+                      const SizedBox(
+                        height: 6,
+                      ),
+                      Text("", style: Theme.of(context).textTheme.subtitle2!.copyWith(color: textColor))
+                    ],
```

Comment:
Yes. The next PR will be adding rate bloc to get the conversion

Reply:
Ah i just mentioned this to gaby, |
If instead of empty string, you put "TODO" as the text, its easier for the reader to spot this is a follow up pr. 
And if you forget to do as a follow up, as soon as the app runs, you will notice. 
Just fyi

Reply:
but then it's ugly! ;)